docs/examples/aws_ebs_volume.rb

```diff
@@ -0,0 +1,48 @@
+require 'chef/provisioning/aws_driver'
+
+with_driver 'aws::us-west-2'
+
+aws_key_pair 'ref-key-pair-ebs'
+
+with_machine_options :bootstrap_options => { :key_name => 'ref-key-pair-ebs' }
+
+machine 'ref-machine-1'
+
+# create volume
+aws_ebs_volume 'ref-volume-ebs' do
+  availability_zone 'us-west-2a'
+  size 1
+end
+
+# attach to machine
+aws_ebs_volume 'ref-volume-ebs' do
+  machine 'ref-machine-1'
+  device '/dev/xvdf'
+end
+
+# reattach to different device
+aws_ebs_volume 'ref-volume-ebs' do
+  device '/dev/xvdg'
+end
+
+# TODO: Fix validation issue
```

Comment:
I was dealing with this same issue earlier - I think there is an error in the LWRPBase.  Can you post your full stacktrace for that `Option name must be a kind of String!` error?